R/contingencytables.R

```diff
@@ -752,19 +769,26 @@ ContingencyTables <- function(jaspResults, dataset, options, ...) {
     else
       group <- NULL
 
-    rows                  <- list()
-    row.count             <- list()
-    row.expected          <- list()
-    row.row.proportions   <- list()
-    row.col.proportions   <- list()
-    row.total.proportions <- list()
-    row.proportions       <- list()
-    row.count[["type[counts]"]] <- "Count"
+    rows                          <- list()
+    row.count                     <- list()
+    row.expected                  <- list()
+    row.row.proportions           <- list()
+    row.col.proportions           <- list()
+    row.total.proportions         <- list()
+    row.proportions               <- list()
+    row.unstandardized.residuals  <- list()
+    row.pearson.residuals         <- list()
+    row.standardized.residuals    <- list()
+    row.count[["type[counts]"]]   <- "Count"
 
     if (ready) {
-      expected.matrix <- try({
-        stats::chisq.test(counts.matrix, correct = FALSE)$expected
+
+      chisqResults <- try({
+        stats::chisq.test(counts.matrix, correct = FALSE)
       })
+
+      expected.matrix <- chisqResults$expected
+
       if (isTryError(expected.matrix)) {
         expected.matrix    <- counts.matrix
         expected.matrix[,] <- ""
```

Comment:
Here the flow of try checks is somewhat in incorrect order. Consider the situation when `isTryError(chisqResults)` is `TRUE`. Then you try to extract `expected.matrix` regardless and you will get an ugly crash of the analysis.

How about

```suggestion

      chisqResults <- try({
        stats::chisq.test(counts.matrix, correct = FALSE)
      })

      if (isTryError(chisqResults)) {
        expected.matrix    <- counts.matrix
        expected.matrix[,] <- ""
      } else {
        expected.matrix <- chisqResults$expected
      }
```